### DISCUSSION TITLE
Confused on Development steps

### DISCUSSION THREAD

**Discussion by crosstowntraffic** [mastodon/mastodon#16802] [Q&A]
I already have an instance of Mastadon running at linode.com inside an ubuntu 20 installation. It is accessible at a domain url -- so I have some familiarity with Mastadon. But this was installed via a Cloudron installation inside my virtual web server that had a Cloudron specific Mastadon auto-install image.

Now I want to make some stylistic page changes in the Mastadon default instances.
So I want to do a localhost Dev installation on my MacPro (pre M1 Mac chip).
**However, the following leading paragraph confuses me -- "You can follow the pre-requisites instructions from the production guide..."**

Does this mean _**FIRST go to that page and begin the process described there ?**_

Or does it mean - _**" You can follow the process on the pre-requisites page, if you want -- HOWEVER, you can also stay on the page you are currently reading and follow the process we describe in the following paragraphs . . ." ?**_
https://docs.joinmastodon.org/dev/setup/

> Pre-requisites
You can follow the pre-requisites instructions from the production guide, but do not create a mastodon user. You also don't have to install nginx, certbot and python-certbot-nginx as the development environment brings its own webserver. Setting up and running a development environment has been proven successful over WSL2 as well if you are on Windows.

**Comment by ClearlyClaire** [comment]
Basically it means you follow most of https://docs.joinmastodon.org/admin/install/ with regards to installing dependencies (except `nginx`, `certbot` and `python-certbot-nginx` are probably not going to be useful)

**Reply by crosstowntraffic** [reply]
Thanks so much!
I like the features of your fork from what I saw of the website installation which is why I decided, as I'm learning "stacks", to follow your version.
May I ask why you decided to switch to Ruby?

Since I saw that in your code I also installed Ruby in my Ubuntu 20 VPS server on my localhost machine.
Are you still using PostgreSQL server database to save & read Mastadon data?

Thank you, CC.

**Reply by ClearlyClaire** [reply]
I'm a bit confused here. I have not switched to Ruby, Mastodon has always been mostly written in Ruby, and glitch-soc does not deviate much from upstream.

**Reply by crosstowntraffic** [reply]
Thank you for taking the time to respond & clear up my mis-understanding!
